plugins/gcp/fix_plugin_gcp/resources/aiplatform.py

```diff
@@ -0,0 +1,3376 @@
+    api_spec: ClassVar[GcpApiSpec] = GcpApiSpec(
+        service="aiplatform",
+        version="v1",
+        service_with_region_prefix=True,
+        accessors=["projects", "locations", "tuningJobs"],
+        action="list",
+        request_parameter={"parent": "projects/{project}/locations/{region}"},
+        request_parameter_in={"project", "region"},
+        response_path="tuningJobs",
+        response_regional_sub_path=None,
+    )
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("name").or_else(S("id")).or_else(S("selfLink")),
+        "tags": S("labels", default={}),
+        "name": S("name"),
+        "ctime": S("creationTimestamp"),
+        "description": S("description"),
+        "link": S("selfLink"),
+        "label_fingerprint": S("labelFingerprint"),
+        "deprecation_status": S("deprecated", default={}) >> Bend(GcpDeprecationStatus.mapping),
+        "base_model": S("baseModel"),
+        "create_time": S("createTime"),
+        "encryption_spec": S("encryptionSpec", "kmsKeyName"),
+        "end_time": S("endTime"),
+        "rpc_error": S("error", default={}) >> Bend(GcpGoogleRpcStatus.mapping),
+        "experiment": S("experiment"),
+        "start_time": S("startTime"),
+        "state": S("state"),
+        "supervised_tuning_spec": S("supervisedTuningSpec", default={})
+        >> Bend(GcpVertexAISupervisedTuningSpec.mapping),
+        "tuned_model": S("tunedModel", default={}) >> Bend(GcpVertexAITunedModel.mapping),
+        "tuned_model_display_name": S("tunedModelDisplayName"),
+        "tuning_data_stats": S("tuningDataStats", default={}) >> Bend(GcpVertexAITuningDataStats.mapping),
+        "update_time": S("updateTime"),
+    }
+    base_model: Optional[str] = field(default=None)
+    create_time: Optional[datetime] = field(default=None)
+    encryption_spec: Optional[str] = field(default=None)
+    end_time: Optional[datetime] = field(default=None)
+    rpc_error: Optional[GcpGoogleRpcStatus] = field(default=None)
+    experiment: Optional[str] = field(default=None)
+    start_time: Optional[datetime] = field(default=None)
+    state: Optional[str] = field(default=None)
+    supervised_tuning_spec: Optional[GcpVertexAISupervisedTuningSpec] = field(default=None)
+    tuned_model: Optional[GcpVertexAITunedModel] = field(default=None)
+    tuned_model_display_name: Optional[str] = field(default=None)
+    tuning_data_stats: Optional[GcpVertexAITuningDataStats] = field(default=None)
+    update_time: Optional[datetime] = field(default=None)
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if tuned_model := self.tuned_model:
+            if endpoint := tuned_model.endpoint:
+                builder.add_edge(self, reverse=True, clazz=GcpVertexAIEndpoint, name=endpoint)
+
+
+resources: List[Type[GcpResource]] = [
+    GcpVertexAIModel,
+    GcpVertexAIDataset,
+    # GcpVertexAIDatasetVersion, : collected via GcpVertexAIFeatureGroup
```

Comment:
can we uncomment them? So we can get the list of resources. 
It is fine if the resources are collected via another resource.

Reply:
We use method `collect_resources` on each of commented resources. for example:
```python
@define(eq=False, slots=False)
class GcpVertexAIDataset(AIPlatformRegionFilter, GcpResource):
    @classmethod
    def collect(cls: Type[GcpResource], raw: List[Json], builder: GraphBuilder) -> List[GcpResource]:
        # Additional behavior: iterate over list of collected GcpVertexAIDataset and for each:
        # - collect related GcpVertexAIDatasetVersion
        result: List[GcpResource] = super().collect(raw, builder)  # type: ignore
        dataset_ids = [dataset.id for dataset in cast(List[GcpVertexAIDataset], result)]
        for dataset_id in dataset_ids:
            builder.submit_work(GcpVertexAIDatasetVersion.collect_resources, builder, parent=dataset_id)

        return result
```
If we uncomment these sub-resources, then collector will use their `api_spec` and will not be able to collect the data.

Reply:
Mhh I am sure that would be solvable - it is not that important to block this PR.